### PR TITLE
adaptors_task_AMarmusevich

### PR DESCRIPTION
**Pull request by marmusevich**
[==========] Running 9 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 9 tests from Balanced
[ RUN      ] Balanced.Easy
[       OK ] Balanced.Easy (0 ms)
[ RUN      ] Balanced.Subexpressions_Positive
[       OK ] Balanced.Subexpressions_Positive (0 ms)
[ RUN      ] Balanced.OneSymbol
[       OK ] Balanced.OneSymbol (0 ms)
[ RUN      ] Balanced.NotClosedOrOpened
[       OK ] Balanced.NotClosedOrOpened (0 ms)
[ RUN      ] Balanced.AntiCounter
[       OK ] Balanced.AntiCounter (0 ms)
[ RUN      ] Balanced.Empty
[       OK ] Balanced.Empty (0 ms)
[ RUN      ] Balanced.Simple
[       OK ] Balanced.Simple (0 ms)
[ RUN      ] Balanced.AdditionalData
[       OK ] Balanced.AdditionalData (0 ms)
[ RUN      ] Balanced.Subexpression_Negative
[       OK ] Balanced.Subexpression_Negative (0 ms)
[----------] 9 tests from Balanced (1 ms total)

[----------] Global test environment tear-down
[==========] 9 tests from 1 test suite ran. (1 ms total)
[  PASSED  ] 9 tests.

### CODE REVIEW

File: adaptors task/student/src/impl.cpp [hunk 1]
```diff
@@ -1,8 +1,40 @@
...
 const std::vector<std::pair<char, char>> brackets {
     {'(', ')'},
     {'[', ']'},
     {'{', '}'}
 };
+
+bool isValid(const std::string& source)
+{
+    bool ret = true;
+    std::stack<char> tempBrackets;
```
Comment (smay1613) [inline adaptors task/student/src/impl.cpp:16]: Never use "temp" in names: on stack everything is temp

File: adaptors task/student/src/impl.cpp [hunk 2]
```diff
@@ -1,8 +1,40 @@
...
+            {
+                tempBrackets.push(c);
+            }
+            else if (!tempBrackets.empty() && tempBrackets.top() == currentBracket->first)
+            {
+                tempBrackets.pop();
+            }
+            else
+            {
+                ret = false;
```
Comment (smay1613) [inline adaptors task/student/src/impl.cpp:34]: why not simple return?

File: adaptors task/student/src/impl.cpp [hunk 3]
```diff
@@ -1,8 +1,40 @@
...
+    bool ret = true;
+    std::stack<char> tempBrackets;
+    for (const char c : source)
+    {
+        auto currentBracket = std::find_if(brackets.cbegin(), brackets.cend(),
+            [c](const std::pair<char, char>& type) { return c == type.first || c == type.second; } );
+
+        if (currentBracket != brackets.cend())
+        {
+            if (c == currentBracket->first)
```
Comment (smay1613) [inline adaptors task/student/src/impl.cpp:24]: please name this condition with a const bool

File: adaptors task/student/src/impl.cpp [hunk 4]
```diff
@@ -1,8 +1,40 @@
...
+            {
+                tempBrackets.push(c);
+            }
+            else if (!tempBrackets.empty() && tempBrackets.top() == currentBracket->first)
```
Comment (smay1613) [inline adaptors task/student/src/impl.cpp:28]: please name this condition with a const bool

**Review by smay1613 — CHANGES_REQUESTED** [review]

File: adaptors task/student/src/impl.cpp (hunk 2 above)
Comment (marmusevich) [reply adaptors task/student/src/impl.cpp:34]: accept

File: adaptors task/student/src/impl.cpp (hunk 1 above)
Comment (marmusevich) [reply adaptors task/student/src/impl.cpp]: фссузе

File: adaptors task/student/src/impl.cpp (hunk 3 above)
Comment (marmusevich) [reply adaptors task/student/src/impl.cpp]: accept

File: adaptors task/student/src/impl.cpp (hunk 4 above)
Comment (marmusevich) [reply adaptors task/student/src/impl.cpp]: accept

**Pull request closed without merging**